### PR TITLE
Patron2020 88 gateway mqtt lib

### PR DESCRIPTION
**Pull request by Deelay1403**
MQTT function for our gateway. Automatically connects to our broker, but we can set dependencies manually.
Lib is Asynchronitious

### CODE REVIEW

File: gateway/dependencies/MQTT.h [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+*/
+void connlost(void *context, char *cause) {
+    printf("\nConnection lost\n");
+    printf("     cause: %s\n", cause);
+}
+
+/*
+Use this for connect to server
+*/
+void MQTT_connect(const char *host = ADDRESS, const char *clientID = CLIENTID, const char *topic = TOPIC_I, bool isSSL = true, const char *caFile = CAFILE) {
```
Comment (kuba8686) [inline gateway/dependencies/MQTT.h:95]: Something is wrong, when i try to run on our gateway code or by Your sample code i got: "Failed to connect, return code -8".
Due to lib description: A structure parameter does not have the correct eyecatcher and version number.
Can resolve it at the moment.

Comment (Deelay1403) [reply gateway/dependencies/MQTT.h:95]: Have you added CA file? It's specified in 45 line as CAFILE. It should be in .pem format.

Comment (kuba8686) [reply gateway/dependencies/MQTT.h:95]: I tried, not work. Probably doing something wrong. I described everything on slack.

**Comment by kuba8686** [comment]
Topic names should be "embedded_output" and "embedded_input" and should be taken from init_configh.

**Comment by kuba8686** [comment]
Paho lib cause problems (seg fault), when deploy at a gateway with ssl turn on. Usen libmosquitto instead. Please decline and close.

**Comment by Deelay1403** [comment]
Closed due Kuba new solution with Mosquitto

**Pull request closed without merging**